model_compression_toolkit/keras/graph_substitutions/substitutions/linear_collapsing.py

```diff
@@ -0,0 +1,111 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.layers import Conv2D
+from model_compression_toolkit.common import BaseNode
+from model_compression_toolkit.common.graph.graph_matchers import NodeOperationMatcher, \
+    NodeFrameworkAttrMatcher
+from model_compression_toolkit.common.substitutions.linear_collapsing import Conv2DCollapsing
+from model_compression_toolkit.keras.constants import KERNEL, STRIDES, PADDING, DILATIONS, LINEAR, \
+    ACTIVATION, BIAS, USE_BIAS, LAYER_NAME
+
+
+def linear_collapsing_node_matchers():
+    """
+    Function generates matchers for matching:
+    (Conv2D, Conv2D)[activation=linear] -> Conv2D.
+    Returns:
+        Matcher for 2 consecutive linear convolution
+    """
+    first_node = NodeOperationMatcher(Conv2D)
+    second_node = NodeOperationMatcher(Conv2D)
+    activation_linear = NodeFrameworkAttrMatcher(ACTIVATION, LINEAR)
+    first_node = first_node & activation_linear
+    return first_node, second_node
+
+
+def conv2d_collapsing_fn(first_node: BaseNode,
+                         second_node: BaseNode,
+                         kernel_str: str,
+                         bias_str: str):
+    """
+    Args:
+        first_node: First layer node to collapse to second layer node
+        second_node: Second layer node
+        kernel_str: The framework specific attribute name of the convolution layer's weight/kernel.
+        bias_str: The framework specific attribute name of the convolution layer's bias.
+    Returns:
+        The modified layer node's weights: kernel, bias
+    """
+    if first_node.type == Conv2D and second_node.type == Conv2D:
+        # Get nodes attributes
+        kernel1 = first_node.get_weights_by_keys(kernel_str)
+        kernel2 = second_node.get_weights_by_keys(kernel_str)
+        bias1 = first_node.get_weights_by_keys(bias_str)
+        bias2 = second_node.get_weights_by_keys(bias_str)
+        strides1 = first_node.framework_attr[STRIDES]
+        strides2 = second_node.framework_attr[STRIDES]
+
+        # Generate identity input
+        kx, ky = kernel1.shape[0] + kernel2.shape[0] - 1, kernel1.shape[1] + kernel2.shape[1] - 1
+        x_pad, y_pad = 2 * kx - 1, 2 * ky - 1
+        in_tensor = tf.eye(kernel1.shape[2])
+        in_tensor = tf.expand_dims(tf.expand_dims(in_tensor, 1), 1)
+        in_tensor = tf.pad(in_tensor, paddings=[[0, 0],
+                                                [int(np.ceil((x_pad - 1) / 2)), int(np.floor((x_pad - 1) / 2))],
+                                                [int(np.ceil((y_pad - 1) / 2)), int(np.floor((y_pad - 1) / 2))],
+                                                [0, 0]])
+
```

Comment:
I see that data_format is const in our DEFAULT_KERAS_INFO and the validate_output_channel_consistency can't handle with ChannelAxis.NCHW. Even if all our layer are NCHW the validation fails because DEFAULT_KERAS_INFO: NHWC

Reply:
I will add skip and warning to that case